api/src/main/java/org/openmrs/validator/PersonNameValidator.java

```diff
@@ -164,4 +179,22 @@ private String getFieldKey(String field, boolean arrayInd, boolean testInd) {
 		return testInd ? field : arrayInd ? "names[0]." + field : "name." + field;
 	}
 	
+	/*
+	 * Checks that the given {@link PersonName} field has no trailing and leading spaces.
+	 * @param value the value name.
+	 * @return containsSpace
+	 * @should return true if given field contains trailing or leading space(s), otherwise false.
+	 */
+	private boolean startsOrEndsWithWhitespace(String value) {
+		boolean startsOrEndsWithWhitespace = false;
+		int lastIndex = value.length() - 1;
+		if (value.charAt(0) == ' ') {
+			startsOrEndsWithWhitespace = true;
+		} else if (Character.isWhitespace(value.charAt(lastIndex))) {
```

Comment:
this also contains file separator, is this what you want? if not maybe thats better
https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isSpaceChar(char)